packages/checkout/widgets-lib/src/widgets/add-funds/views/AddFunds.tsx

```diff
@@ -230,6 +231,31 @@ export function AddFunds({
   //   console.log('handle review click');
   // }, []);
 
+  useEffect(() => {
```

Comment:
Doing this calculation in react, reduces the testability and performance of the code.

I suggest you move these to a `getFiatAmount` utility fn:

```export function getFiatAmount(amount: string, token: TokenInfo, tokens: TokenInfo[]): Number {
  if (!amount || !token?.address || !tokens) {
    return 0;
  }

  const address =
    token.address === "native" ? SQUID_NATIVE_TOKEN : token?.address;

  const toToken = tokens.find(
    (value) =>
      value.address.toLowerCase() === address.toLowerCase() &&
      value.chainId === ChainId.IMTBL_ZKEVM_MAINNET.toString()
  );

  if (!toToken) {
    return 0;
  }

  return Number(amount) * toToken.usdPrice;
}
```
and set the value using a `useMemo` hook.

```
const fiatAmount = useMemo(
  () => getFiatAmount(inputValue, currentToTokenAddress, tokens);
}, [inputValue, currentToTokenAddress, tokens]);
```

Reply:
Done